File: cellab_devkit-1.0 (1)/ruledefs/fractal.js
    /*

        Based on Me-Neither rule, [Margolus&Toffoli87], p.132

    */

    rule.worldtype = 1;          // 2D torus world
    rule.patreq = "square";

    function fractal(oldstate,     nw, n  , ne,
                                   w, self, e,
                                   sw, s  , se) {
        var mem, sum, newSelf;

        mem = (oldstate >> 1) & 1;
        sum = ne + nw + se + sw + mem;
        newSelf = ((sum & 1) == 1) ? 1 : 0;
        
        return (self << 1) | newSelf;
    }

File: references/cellab_devkit-1.0 (1)/rules_java/fractal.java
/*

        Based on Me-Neither rule, [Margolus&Toffoli87], p.132

*/

class Fractal extends ruletable {

    void jcruleModes() {
        setPatternRequest("square");
    }

    int jcrule(int oldstate) {
        int Mem, Sum, NewSelf;

        Mem = BITFIELD(1);
        Sum = ne + nw + se + sw + Mem;
        NewSelf = ((Sum & 1) == 1) ? 1 : 0;
        return (self << 1) | NewSelf;
    }
}

public class fractal {
    public static void main(String args[]) {
        (new Fractal()).generateRuleFile("fractal");
    }
}
